### PR TITLE
Precompute interval range

### PR DESCRIPTION
**Pull request by thomaseizinger**
This makes for a big performance improvement if `range` is called in a tight loop because it removes unnecessary allocations and computations.

### CODE REVIEW

**Review by klochowicz — COMMENTED** [review]
great one!

File: src/interval.rs [hunk 1]
```diff
@@ -15,7 +15,14 @@ const BASE: usize = 2;
...
 /// Binary representation of a price interval.
 #[derive(Clone, Debug, PartialEq)]
-pub struct Digits(BitVec);
+pub struct Digits {
+    inner: BitVec,
+
+    /// The precomputed range represents by these `Digits`.
+    ///
+    /// This is pre-computed for performance reasons.
+    range: RangeInclusive<u64>,
```
Comment (luckysori) [inline src/interval.rs:24]: 👍 I suspected you would end up doing this when I looked at the code a few hours ago.

**Review by luckysori — COMMENTED** [review]
Sorry for my slow code 🙃

**Comment by thomaseizinger** [comment]
> Sorry for my slow code :upside_down_face:

The code isn't slow per se! It is just being called in our old friend the triple loop and that is why it is on the hot path. Impossible to know when you are writing it. I would have 100% written this the same way,

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -18,13 +18,8 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2.3.5
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          override: true
-          toolchain: stable
-          components: clippy
+      - name: Setup rust toolchain
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply .github/workflows/ci.yml:21]: I'll fix it in my clean-up PR!

**Pull request merged**